File: backend/src/main/java/com/back/ourlog/domain/user/repository/UserRepository.java
package com.back.ourlog.domain.user.repository;

import com.back.ourlog.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {

    Optional<User> findByEmail(String email);

    Optional<User> findByProviderAndProviderId(String provider, String providerId);

    Page<User> findByNicknameContainingIgnoreCase(String keyword, Pageable pageable);
}

File: backend/src/main/java/com/back/ourlog/domain/user/repository/UserRepository.kt
package com.back.ourlog.domain.user.repository

import com.back.ourlog.domain.user.entity.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, Int> {

    fun findByEmail(email: String): Optional<User>

    fun findByProviderAndProviderId(provider: String, providerId: String): Optional<User>

    fun findByNicknameContainingIgnoreCase(keyword: String, pageable: Pageable): Page<User>
}